src/booking/containers/booking-home/booking-home.component.ts

```diff
@@ -1,27 +1,64 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
-import { BookingNavigationEvent, NewBooking } from '../../models';
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { Booking, BookingNavigationEvent, NewBooking } from '../../models';
+import { BookingService } from '../../services';
 
 @Component({
   selector: 'exui-booking-home',
-  templateUrl: './booking-home.component.html',
-  styleUrls: ['./booking-home.component.scss']
+  templateUrl: './booking-home.component.html'
 })
-export class BookingHomeComponent implements OnInit {
+export class BookingHomeComponent implements OnInit, OnDestroy {
 
   @Input() public newBooking: NewBooking;
 
   @Output() public newBookingChange = new EventEmitter<NewBooking>();
   @Output() public eventTrigger = new EventEmitter();
 
-  constructor() { }
+  public bookingTypeForm: FormGroup;
+  public existingBookings: Booking[];
+  private existingBookingsSubscription: Subscription;
+  private bookingLocationSubscription: Subscription;
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly bookingService: BookingService
+  ) { }
 
   public ngOnInit() {
+    this.bookingTypeForm = this.fb.group({
+      bookingType: new FormControl(null)
+    });
+
+    this.existingBookingsSubscription = this.bookingService.getBookings().subscribe(response => {
+      this.existingBookings = response.bookings;
+      this.assignBookingLocation();
+    });
   }
 
   public onSelectOption(index) {
     this.newBooking.bookingOptionIndex = index;
   }
 
+  public assignBookingLocation() {
+    this.existingBookings.forEach(booking => {
+      this.bookingLocationSubscription = this.bookingService.getBookingLocation(booking.base_location_id).subscribe(location =>
+        booking.locationName = location[0] && location[0].building_location_name ?
```

Comment:
Can this be changed to booking.locationName = location[0]?.building_location_name || null

Reply:
After checking it appears not! Ignore / resolved.